packages/babel-plugin-proposal-decorators/src/transformer-2021-12.ts

```diff
@@ -0,0 +1,946 @@
+}
+
+function addProxyAccessorsFor(
+  element: NodePath<ClassDecoratableElement>,
+  originalKey: t.PrivateName | t.Expression,
+  targetKey: t.PrivateName,
+  isComputed = false,
+): void {
+  const { static: isStatic } = element.node;
+
+  const getterBody = t.blockStatement([
+    t.returnStatement(
+      t.memberExpression(t.thisExpression(), t.cloneNode(targetKey)),
+    ),
+  ]);
+
+  const setterBody = t.blockStatement([
+    t.expressionStatement(
+      t.assignmentExpression(
+        "=",
+        t.memberExpression(t.thisExpression(), t.cloneNode(targetKey)),
+        t.identifier("v"),
+      ),
+    ),
+  ]);
+
+  let getter: t.ClassMethod | t.ClassPrivateMethod,
+    setter: t.ClassMethod | t.ClassPrivateMethod;
+
+  if (originalKey.type === "PrivateName") {
+    getter = t.classPrivateMethod(
+      "get",
+      t.cloneNode(originalKey),
+      [],
+      getterBody,
+      isStatic,
+    );
+    setter = t.classPrivateMethod(
+      "set",
+      t.cloneNode(originalKey),
+      [t.identifier("v")],
+      setterBody,
+      isStatic,
+    );
+  } else {
+    getter = t.classMethod(
+      "get",
+      t.cloneNode(originalKey),
+      [],
+      getterBody,
+      isComputed,
+      isStatic,
+    );
+    setter = t.classMethod(
+      "set",
+      t.cloneNode(originalKey),
+      [t.identifier("v")],
+      setterBody,
+      isComputed,
+      isStatic,
+    );
+  }
+
+  element.insertAfter(setter);
+  element.insertAfter(getter);
+}
+
+function extractProxyAccessorsFor(
+  targetKey: t.PrivateName,
+): t.FunctionExpression[] {
+  return [
+    t.functionExpression(
+      undefined,
+      [],
+      t.blockStatement([
+        t.returnStatement(
+          t.memberExpression(t.thisExpression(), t.cloneNode(targetKey)),
+        ),
+      ]),
+    ),
+    t.functionExpression(
+      undefined,
+      [t.identifier("value")],
+      t.blockStatement([
+        t.expressionStatement(
+          t.assignmentExpression(
+            "=",
+            t.memberExpression(t.thisExpression(), t.cloneNode(targetKey)),
+            t.identifier("value"),
+          ),
+        ),
+      ]),
+    ),
+  ];
+}
+
+const FIELD = 0;
+const ACCESSOR = 1;
+const METHOD = 2;
+const GETTER = 3;
+const SETTER = 4;
+
+const STATIC = 5;
+
+function getElementKind(element: NodePath<ClassDecoratableElement>): number {
+  switch (element.node.type) {
+    case "ClassProperty":
+    case "ClassPrivateProperty":
+      return FIELD;
+    case "ClassAccessorProperty":
+      return ACCESSOR;
+    case "ClassMethod":
+    case "ClassPrivateMethod":
+      if (element.node.kind === "get") {
+        return GETTER;
+      } else if (element.node.kind === "set") {
+        return SETTER;
+      } else {
+        return METHOD;
+      }
+  }
+}
+
+function generateLocalVarId(path: NodePath, name: string): t.Identifier {
+  const varId = path.scope.generateUidIdentifier(name);
+  path.scope.parent.push({ id: varId });
+  return t.cloneNode(varId);
+}
```

Comment:
You can use `path.scope.parent.generateDeclaredUidIdentifier` instead, which does exactly what you need.